code/parse/sexp.cpp

```diff
@@ -11504,6 +11504,93 @@ void sexp_end_campaign(int n)
 	}
 }
 
+void set_subsys_strength_and_maybe_ancestors(ship *shipp, ship_subsys *ss, polymodel *pm, const int *assign_percent, const int *repair_percent, const int *sabotage_percent, bool do_submodel_repair, bool repair_ancestors)
+{
+	Assertion(shipp != nullptr && ss != nullptr, "Ship and subsys must not be null!");
+	Assertion(assign_percent != nullptr || repair_percent != nullptr || sabotage_percent != nullptr, "Either assign_percent or repair_percent or sabotage_percent must not be null!");
+
+	bool originally_zero = (ss->current_hits <= 0 || ss->submodel_info_1.blown_off || ss->submodel_info_2.blown_off);
+
+	if (assign_percent != nullptr)
+	{
+		int percentage = *assign_percent;
+		Assertion(percentage >= 0 && percentage <= 100, "Percentage must be in range [0, 100]");
+
+		// maybe blow up subsys
+		if (ss->current_hits > 0 && percentage < 1)
+			do_subobj_destroyed_stuff(shipp, ss, nullptr);
+
+		// assign the hitpoints
+		ss->current_hits = ss->max_hits * ((float)percentage / 100.0f);
+	}
+	else if (repair_percent != nullptr)
+	{
+		int percentage = *repair_percent;
+		Assertion(percentage > 0, "Repair percentage must be more than zero!");
+
+		// repair the hitpoints
+		float repair_hits = ss->max_hits * ((float)percentage / 100.0f);
+		ss->current_hits += repair_hits;
+		if (ss->current_hits > ss->max_hits)
```

Comment:
No -- quite the opposite, in my experience.  Plenty of places where `std::min` or `std::max` is used would benefit from being rewritten as if() blocks, because you need to read the statement two or three times to make sure the logic is correct.  Here, there is no ambiguity; it is clear at first glance.

Reply:
fair enough